lib/spack/docs/packaging_guide.rst

```diff
@@ -373,6 +373,102 @@ some examples:
 In general, you won't have to remember this naming convention because
 :ref:`spack-create` and :ref:`spack-edit` handle the details for you.
 
+-----------------
+Trusted Downloads
+-----------------
+
+Spack verifies that the source code it downloads is not corrupted or
+compromised; or at least, that it is the same version the author of
+the Spack package saw when the package was created.  If Spack uses a
+download method it can verify, we say the download method is
+*trusted*.  Trust is important for *all downloads*: Spack
+has no control over the security of the various sites from which it
+downloads source code, and can never assume that any particular site
+hasn't been compromised.
+
+Trust is established in different ways for different download method.
+For the most common download method --- a single-file tarball --- the
+tarball is checsummed.  Git downloads using ``commit=`` are trusted
+implicitly, as long as a hash is specified.
+
+Spack also provides untrusted download methods: tarball URLs may be
+supplied without a checksum, or Git downloads my specify a branch or
+tag instead of a hash.  If the user does not control or trust the
+source an untrusted download, it is a security risk.  Unless otherwise
+specified by the user for special cases, Spack should by default use
+*only* trusted download methods.
+
+Unfortunately, Spack does not currently provide that guarantee.  It
+does provide the following mechanisms for safety:
+
+#. By default, Spack will only install a tarball package if it has a
+   checksum and that checksum matches.  You can override this with
+   ``spack install --no-checksum``.
+
+#. Numeric versions are almost always tarball downloads, whereas
+   non-numeric versions not named ``develop`` frequently download
+   untrusted branches or tags from a version control system.  As long
+   as a package has at least one numeric version, and no non-numeric
+   version named ``develop``, Spack will prefer it over any
+   non-numeric versions.
```

Comment:
The "develop" vs. "non-numeric version" distinction here has me confused. My understanding was that "develop" was untrusted just like any other "non-numeric version". The only difference was that Spack was smart enough to understand that "develop" was newer, but should not be preferred during concretization. @davydden Is this right?